### PR TITLE
Improved PostgreSQL support, squash some bugs and general maintenance

### PR DESCRIPTION
**Pull request by gronostajo**
I've done a number of changes:

- BC break: better Postgres function names. I have no idea what `GT_GT` does, but I more-or-less know what `JSON_GET_TEXT` does.
-  Added Symfony bundle + bundle-based Postgres config. All Postgres functions can now be enabled in Symfony apps just by importing `@DoctrineJsonFunctionsBundle/Resources/config/postgres_functions.yml` into config. No extra coding necessary.
- Added JSONB operators: `?`, `?|`, `?&`, `@>`, `<@`
- Fixed issues with negative indices. It appears that passing numeric indices as text works just as well for regular numbers, but doesn't break for negatives.
- Added support for parametrizing queries that use JSON functions.
- Extracted repeating code

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
 		"php": ">=5.3.2"
 	},
 	"require-dev": {
-		"doctrine/orm": "~2.1"
+		"doctrine/orm": "~2.1",
+		"symfony/framework-bundle": "^3.3"
```
Comment (Hikariii) [inline composer.json]: I'd like to keep this framework independent, so please remove this line.

File: src/DoctrineJsonFunctionsBundle/Resources/config/postgres_functions.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+doctrine:
```
Comment (Hikariii) [inline src/DoctrineJsonFunctionsBundle/Resources/config/postgres_functions.yml]: The same here. I'd like to see this removed completely.
It's up to the framework user to configure this.

**Comment by gronostajo** [comment]
Done.

Here's [a gist with the Symfony code](https://gist.github.com/gronostajo/b345b51ad60dd701bf0f538e7e0ecb9c), in case someone needs it (and finds it) at some point.

**Comment by fracz** [comment]
Can you release a new version with this PR merged, please?

**Pull request merged**